pkg/api/types.go

```diff
@@ -265,8 +265,8 @@ type GitRepoVolumeSource struct {
 // The contents of the target Secret's Data field will be presented in a volume
 // as files using the keys in the Data field as the file names.
 type SecretVolumeSource struct {
-	// Reference to a Secret
-	Target ObjectReference `json:"target"`
+	// Name of the secret in the pod's namespace to use
+	Target string `json:"target"`
```

Comment:
I don't know that it has to be target specifically, but the semantics of secrets (and probably claims?) are such that no secret should be visible outside its namespace.  It's also not polymorphic, although I suppose you could write a volume plugin that allowed that.  In 99% of use cases only Name should be set.  Not sure whether that is enough to tip it into not being an object reference.


> On Mar 23, 2015, at 4:38 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/types.go:
> 
> > @@ -265,8 +265,8 @@ type GitRepoVolumeSource struct {
> >  // The contents of the target Secret's Data field will be presented in a volume
> >  // as files using the keys in the Data field as the file names.
> >  type SecretVolumeSource struct {
> > -	// Reference to a Secret
> > -	Target ObjectReference `json:"target"`
> > +	// Name of the secret in the pod's namespace to use
> > +	Target string `json:"target"`
> Is this the general form we want such cross-object references to take? Was there discussion leading up to this that we can document?
> 
> —
> Reply to this email directly or view it on GitHub.
>